src/usps_processor/parse.clj

```diff
@@ -33,24 +34,117 @@
   "Takes the time in the USPS string format, and a timezone id like
    'America/New_York', and returns a DateTime in UTC for that time."
   [time-string timezone-id]
-  (let [timezone (timezone-id->timezone timezone-id)
-        formatter (format/formatter usps-time-format-pattern timezone)
-        time-in-zone (format/parse formatter time-string)]
-    (time/to-time-zone time-in-zone time/utc)))
+  (try
+    (let [timezone (timezone-id->timezone timezone-id)
+          formatter (format/formatter usps-time-format-pattern timezone)
+          time-in-zone (format/parse formatter time-string)]
+      (time/to-time-zone time-in-zone time/utc))
+    (catch Exception ex
+      (log/warn "Exception parsing scan time" ex))))
+
+(defn datetime->date
+  "Converts a datetime to date, or returns nil if datetime is nil"
+  [datetime]
+  (when datetime
+    (coerce/to-date datetime)))
+
+(defn ->imb-data
+  "Combines the structure-digits and routing code and
+   splits them into the possible mailer-ids and
+   serial numbers"
+  [structure-digits routing-code]
+  (when (and structure-digits
+             (= 20 (count structure-digits)))
+    (imb/split-structure-digits
+     (str structure-digits routing-code))))
```

Comment:
That's the reason there's a check on the count of structure-digits, since the code it calls does all kinds of substring calls with very specific counts. I don't believe `imb/split-structure-digits` is capable of returning nil, nor throwing an exception so long as structure-digits is 20 characters long the count check assures.

If the whole `->imb-data` returns nil, that gets caught by the schema validation.